osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -222,12 +222,13 @@ public double GetScore(ScoringMode mode, int maxCombo, double accuracyRatio, dou
                 case ScoringMode.Standardised:
                     double accuracyScore = accuracyPortion * accuracyRatio;
                     double comboScore = comboPortion * comboRatio;
-
                     return (max_score * (accuracyScore + comboScore) + getBonusScore(statistics)) * scoreMultiplier;
 
                 case ScoringMode.Classic:
-                    // should emulate osu-stable's scoring as closely as we can (https://osu.ppy.sh/help/wiki/Score/ScoreV1)
-                    return getBonusScore(statistics) + (accuracyRatio * Math.Max(1, maxCombo) * 300) * (1 + Math.Max(0, (comboRatio * maxCombo) - 1) * scoreMultiplier / 25);
+                    // This gives a similar feeling to osu!stable scoring (ScoreV1) while keeping classic scoring as only a constant multiple of standardised scoring.
+                    // The invariant is important to ensure that scores don't get re-ordered on leaderboards between the two scoring modes.
+                    double scaledStandardised = GetScore(ScoringMode.Standardised, maxCombo, accuracyRatio, comboRatio, statistics) / max_score;
+                    return Math.Pow(scaledStandardised * (maxCombo + 1), 2) * 18;
```

Comment:
I added 1 to this to fix the only-bonus-score case. It'll change scores by a bit but I don't think it's relevant enough to reconsider the `18` multiplier.

Reply:
The difference in the inner term will not exceed `2 * maxCombo + 1`, so in the large scale of things this should not have a significant impact on the final numbers.